_data-prepper/peer-forwarder.md

```diff
@@ -0,0 +1,181 @@
+    "Statement": [
+        {
+            "Sid": "CloudMapPeerForwarder",
+            "Effect": "Allow",
+            "Action": "servicediscovery:DiscoverInstances",
+            "Resource": "*"
+        }
+    ]
+}
+```
+
+
+## Configuration
+
+The following table provides optional configuration values.
+
+
+| Value | Description |
+| ----  | ----------- |
+| `port` | An `int` between 0 and 65535 represents the port that the Peer Forwarder server is running on. Default value is `4994`. |
+| `request_timeout` | An `int` representing the request timeout duration in milliseconds for the Peer Forwarder HTTP server. Default value is `10000`. |
+| `server_thread_count` | An `int` representing the number of threads used by the Peer Forwarder server. Default value is `200`.|
+| `client_thread_count` | An `int` representing the number of threads used by the Peer Forwarder client. Default value is `200`.|
+| `maxConnectionCount`  |  An `int` representing the maximum number of open connections for the Peer Forwarder server. Default value is `500`. |
+| `discovery_mode` | A `String` representing the peer discovery mode to be used. Allowable values are `local_node`, `static`, `dns`, and `aws_cloud_map`. Defaults to `local_node`, which processes events locally. |
+| `static_endpoints` |  A `list` containing the endpoints of all Data Prepper instances. Required if `discovery_mode` is set to `static`. |
+|  `domain_name` | A `String` representing the single domain name to query DNS against. Typically, used by creating multiple [DNS A records](https://www.cloudflare.com/learning/dns/dns-records/dns-a-record/) for the same domain. Required if `discovery_mode` is set to `dns`. |
+| `aws_cloud_map_namespace_name`  | A `String` representing the AWS Cloud Map namespace when using AWS Cloud Map service discovery. Required if `discovery_mode` is set to `aws_cloud_map`.  |
+| `aws_cloud_map_service_name` |  A `String` representing the AWS Cloud Map service when using AWS Cloud Map service discovery. Required if `discovery_mode` is set to `aws_cloud_map`. |
+| `aws_cloud_map_query_parameters`  |  A `Map` of key-value pairs used to filter the results based on the custom attributes attached to an instance. Only instances that match all the specified key-value pairs are returned. |
+| `buffer_size` |  An `int` representing the maximum number of unchecked records the buffer accepts (the number of unchecked records equals the number of records written into the buffer plus the number of in-flight records not yet checked by the Checkpointing API). Default is `512`. |
+| `batch_size` |  An `int` representing max number of records the buffer returns on read. Default is `48`. |
+|  `aws_region` |  A `String` represents the AWS Region to use `ACM`, `S3` or `AWS Cloud Map`. Required if `use_acm_certificate_for_ssl` is set to `true` or `ssl_certificate_file`,  and `ssl_key_file` is set to the `AWS S3` path, or if `discovery_mode` is set to `aws_cloud_map`. |
```

Comment:
@dlvenable Need some clarification on this one as well.

Reply:
@cwillum There are three conditions here
`aws_region` is required if any of the following is true
1. `use_acm_certificate_for_ssl` is set to `true`
2.  `ssl_certificate_file` and `ssl_key_file` is set to the AWS S3 path
3. `discovery_mode` is set to `aws_cloud_map`.

Reply:
> ssl_certificate_file and ssl_key_file is set to the AWS S3 path

One additional point of clarification. This means that the administrator entered a value which is an S3 URI - for example `s3://mybucket/path/to/public.cert`.

Reply:
Maybe something like this, then:
"`aws_region` is required when any of the following conditions are met: 1) `use_acm_certificate_for_ssl` is set to true; 2) `ssl_certificate_file` and `ssl_key_file` are set to the AWS S3 URI (for example, `s3://mybucket/path/to/public.cert`); or 3) `discovery_mode` is set to `aws_cloud_map`.

For the second condition, I just want to be sure that the two properties `ssl_certificate_file` and `ssl_key_file` are set to the same URI (not just one or the other).

Reply:
> For the second condition, I just want to be sure that both the two properties ssl_certificate_file and ssl_key_file specify the same URI (not just one or the other). Thanks for the feedback.

These are distinct properties. I believe it is possible for one to be an S3 URI and the other a local file path.

So I might say:

> 2) ssl_certificate_file or ssl_key_file are set to an AWS S3 URI (for example, s3://mybucket/path/to/public.cert); 

Also, note that it I think you should use the indefinite article here - "set to an AWS S3 URI".

Reply:
@dlvenable @asifsmohammed 
OK, when one or the other specifies an AWS S3 URI. Thanks for the clarification. 
Also, In our style guide, we decided to call key/value pairs "settings". So I think we can go with this ...

"`aws_region` is required when any of the following conditions are met: 1) the `use_acm_certificate_for_ssl` setting is `true`; 2) either `ssl_certificate_file` or `ssl_key_file` specifies an AWS S3 URI (for example, s3://mybucket/path/to/public.cert); or 3) the `discovery_mode` setting is `aws_cloud_map`.

@carolxob In running text it would be clearer to set these conditions in bullet points. Since this definition falls in a table, we might run this formatting option by the team to get some opinions on whether this style works in a table. If not, we can still list them as above.